include/ygm/container/detail/bag_impl.hpp

```diff
@@ -52,6 +54,26 @@ class bag_impl {
     m_local_bag.swap(s.m_local_bag);
   }
 
+  void local_shuffle(ygm::default_random_engine<> r) {
```

Comment:
This should be templated on the random_engine. Should have a version that doesn't take in a random_engine at all and creates a ygm::default_random_engine using the bag's m_comm to pass through to the templated version.

These random_engines should also be passed by reference.

Reply:
Thanks for the feedback Trevor. Do you want me to template on the ygm::detail::random_engine or on ygm::default_random_engine? The main reason I ask is because the ygm::default_random_engine already templates random_engine with the simple_offset for seeding. Unless I need to also provide a template for the offset in the bag shuffle functions as well it might be better to template on the default_random_engine.

Reply:
A version that doesn't receive a random engine should use the default_random_engine with std::mt_19937 as its underlying generator. The templated version should be completely generic; it shouldn't be assuming it's using a generator from YGM.

Reply:
Sounds good. Will do.